ios/NativeViews/IpfsImageSource.h

```diff
@@ -0,0 +1,35 @@
+#import <Foundation/Foundation.h>
+
+#import <React/RCTConvert.h>
+
+/**
+ * Object containing an image hash path and associated metadata.
+ */
+@interface IpfsImageSource : NSObject
```

Comment:
This is the class that models the backing data for `IpfsImage`. Looks just like `RCTImage` source, but with the `NSURLRequest` swapped out for a `NSString` for the hash path.